informer/src/informer_service/informer_service_handler.rs

```diff
@@ -14,33 +14,39 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use crate::{informer_notify, Connection, InformerEventSender};
+use crate::{informer_notify, Connection, EventTags, Events, InformerEventSender};
+use ccore::{BlockChainTrait, BlockId, Client, EngineInfo};
 use crossbeam::Receiver;
 use crossbeam_channel as crossbeam;
-use std::thread::spawn;
-
-#[derive(Clone)]
-pub enum EventTags {
-    PeerAdded,
-}
+use crpc::v1::Block as RPCBlock;
+use leaky_bucket::LeakyBucket;
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+use tokio::runtime::Runtime;
+use tokio::task;
 
 #[derive(Serialize)]
-pub enum Events {
-    PeerAdded(String, String, usize),
+pub enum ColdEvents {
+    None,
+    BlockGeneration(Box<RPCBlock>),
+    SendingDone(String),
 }
 
 pub struct InformerService {
-    connections: Vec<Connection>,
+    connections: Vec<Arc<Connection>>,
```

Comment:
I think because I am using the connection in different threads, Arc help to avoid cloning it.

Reply:
Why I said "Remove clone" is that there was no need to use "clone". In most cases, you can use a reference instead of calling clone.
Using `Arc` to avoid cloning is awkward.

You should consider whether you will use Arc or not when you design the `Connection` type.

What I feel awkward is this point. If you want to make only one Connection instance for real connection, you can make `Connection` not Clonable and only use it with `Arc`.

If you want to make the ColdEventGenerator and the InformerService have its own Connection object for the same real connection, you should not use `Arc` and make the `Connection` type Clonable.